### DISCUSSION TITLE
Is there any way to get back usage tokens from responses?

### DISCUSSION THREAD

**Discussion by bradenkinard** [567-labs/instructor#172] [Q&A] [answered]
Really liking the library, its very slick. However, I'd like to be able to get back the usage tokens OpenAI returns in the response, or more generally, the metadata beyond just the response message. The Pydantic validation and automatic retries this library offers is incredibly useful, but once that is done, having access to the raw OpenAI response would also be nice, at least in my case.

**Comment by jxnl** [comment, marked as answer]
its not very documented but `Model.resp._raw_response` is the raw response. I'll improve the documentation there:

<img width="796" alt="Screenshot 2023-11-13 at 5 24 05 PM" src="https://github.com/jxnl/instructor/assets/4852235/e865d286-e6a1-4302-a3a9-eee4b5368b38">

**Reply by bradenkinard** [reply]
Perfect! This is exactly what I needed. Thanks!
